src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -341,6 +341,10 @@ def loadBlocks(self):
             block = DBSBufferBlock(name=blockInfo['block_name'],
                                    location=blockInfo['origin_site_name'],
                                    datasetpath=blockInfo['datasetpath'])
+
+            parents = self.getStepChainParentage(blockInfo['datasetpath'])
+            block.addDatasetParents(parents)
```

Comment:
`addParentDataset`?

Reply:
And when it's not a StepChain block, the DAO will return an empty list or something like that, shouldn't we then skip this `block.addDatasetParents` call such that we leave the correct parentage information for non-stepchain requests?